spec/acceptance/suites/default/redis_multi_instances_one_host_spec.rb

```diff
@@ -34,29 +50,23 @@ class { 'redis':
     it { is_expected.to be_installed }
   end
 
-  describe service('redis-server-redis1') do
-    it { is_expected.to be_running }
-  end
-
-  describe service('redis-server-redis2') do
-    it { is_expected.to be_running }
-  end
-
-  describe file("#{config_path}/redis-server-redis1.conf") do
-    its(:content) { is_expected.to match %r{port 7777} }
-  end
+  instances.each do |instance|
+    describe file("/etc/systemd/system/redis-server-#{instance}.service"), if: (fact('service_provider') == 'systemd') do
+      its(:content) { is_expected.to match %r{redis-server-#{instance}.conf} }
+    end
 
-  describe file("#{config_path}/redis-server-redis2.conf") do
-    its(:content) { is_expected.to match %r{port 8888} }
-  end
+    describe service("redis-server-#{instance}") do
+      it { is_expected.to be_running }
+    end
 
-  context 'redis should respond to ping command' do
-    describe command('redis-cli -h 127.0.0.1 -p 7777 ping') do
-      its(:stdout) { is_expected.to match %r{PONG} }
+    describe file("#{config_path}/redis-server-#{instance}.conf") do
+      its(:content) { is_expected.to match %r{port #{instance}} }
     end
 
-    describe command('redis-cli -h 127.0.0.1 -p 8888 ping') do
-      its(:stdout) { is_expected.to match %r{PONG} }
+    context "redis instance #{instance} should respond to ping command" do
+      describe command("redis-cli -h #{fact('networking.ip')} -p #{instance} ping") do
```

Comment:
Can this use localhost instead of the IP?

Reply:
No, because we are testing the example from the README, which uses `bind => $facts['networking']['ip']`. Hence it would not work with localhost.